proxy/coco/src/peer/run_state.rs

```diff
@@ -157,12 +157,25 @@ pub enum AnnounceInput {
 /// Requests from the peer control.
 #[derive(Debug)]
 pub enum ControlInput {
-    /// Request a project request.
-    ProjectRequest(RadUrn, oneshot::Sender<Option<SomeRequest<Instant>>>),
-    /// Request the list of project requests.
-    ProjectRequests(oneshot::Sender<Vec<SomeRequest<Instant>>>),
     /// New status.
     Status(oneshot::Sender<Status>),
+
+    /// Cancel an ongoing project search.
+    CancelRequest(
```

Comment:
What about having a `Search` enum here which is embedded in `ControlInput`? Just to separate them out cognitively.

Reply:
Tried that, not enough variants for now, we can come to that later.